activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -325,9 +325,9 @@ def column(name, type, options = {})
           raise ArgumentError, "you can't redefine the primary key column '#{name}'. To define a custom primary key, pass { id: false } to create_table."
         end
 
-        index_options = options.delete(:index)
```

Comment:
Given that, performance-wise, you are implicitly duping the hash in the `expect` call later, perhaps it's clearer to do `options = options.dup` at beginning of method, and then can safely mutate them later as before.